tchannel/tornado/connection.py

```diff
@@ -452,12 +466,26 @@ def serve(self, handler):
         """
         assert handler, "handler is required"
         assert self._loop_running, "Finish the handshake first"
-
         while not self.closed:
             message = yield self.await()
-
             try:
-                handler(message, self)
+                if self.draining:
```

Comment:
You can't just remove all handlers. Think about the purpose.
1. there is exempt function exist, which mean some of the service will still be accepted during draining.
2. It needs to be able to relaunch after stop to solve the global test instance issue. If I remove all handlers, there is no way it can be relaunch